docs/scripts/gen-faq.js

```diff
@@ -0,0 +1,151 @@
+// read.js
+const fs = require('fs')
+const yaml = require('js-yaml')
+const { Remarkable } = require('remarkable')
+const path = require('path')
+const yamlPath = path.resolve('./docs/faq.yaml')
+
+// Generating FAQ.mdx
+
+if (!fs.existsSync(yamlPath)) {
+  //file exists
+  console.warn('.yaml File does not exists, skipping generating FAQ')
+  return 0
+}
+
+let faqYaml = fs.readFileSync(yamlPath, 'utf8')
+let faq = yaml.load(faqYaml)
+
+const tags = Array.from(new Set(faq.map(({ tags }) => tags).flat(1)))
+
+let data = `---
+id: faq
+title: Frequently Asked Questions (FAQ)
+---
+
+
+
+import {Question, Faq} from '@theme/Faq'
+
+<Faq tags="${tags.join(' ')}"/>
+<br/><br/>
+
+`
+const md = new Remarkable()
+faq.forEach((el) => {
+  const react_tags = el.tags.map((tag) => {
+    return tag + '_src-theme-'
+  })
+  data += `<Question tags="question_src-theme- ${react_tags.join(' ')}">\n`
+  data += `${el.tags.map(({ tag }) => '#' + tag).join(' ')} <br/>\n`
+  data += md.render(`**Q**: ${el.q}`)
+  data += md.render(`**A**: ${el.a}\n`)
+  if (el.context) {
+    data += md.render(`context: ${el.context}\n`)
+  }
+  data += `</Question>\n\n<br/>`
+})
+
```

Comment:
It would be good to run prettier if you have the file in memory already, similar to https://github.com/ory/kratos/blob/4bb4f679d1fe0a49edb0c0189bb7a2188d4f850d/docs/scripts/config.js#L221

Reply:
Unfortunatly this is a mix of html/markdown and prettier is either not at all formatting html (with the markdown parser) or mixing up the syntax (with the html parser). 
We improved the generated code instead.